.github/workflows/test.yml

```diff
@@ -3,41 +3,137 @@ name: CI
 on: [push, pull_request]
 
 jobs:
-    unix-test:
+    build-test:
         runs-on: ${{ matrix.os }}
         strategy:
             fail-fast: false
             matrix:
                 os: [ubuntu-latest, macos-latest]
+                build: [general, webrender]
 
         steps:
             - uses: actions/checkout@v2
 
+            - name: rust setup
+              id: rust-setup
+              run: |
+                  rustup install $(cat rust-toolchain)
+                  rustup component add rustfmt-preview
+                  echo "::set-output name=week::$(/bin/date -u '+%Yy_%mm_%Ww')"
+                  echo "::set-output name=date::$(/bin/date)"
+
+            - name: rust cache
+              uses: actions/cache@v2
+              with:
+                  path: |
+                      ~/.cargo/registry/cache
+                      ~/.cargo/git
+                      ./rust_src/target
+                  key: ${{ runner.os }}-${{ matrix.build }}-rust-${{ hashFiles('**/rust-toolchain', '**/Cargo.toml*', '**/rustfmt.toml') }}-${{ steps.rust-setup.outputs.week }}
+                  restore-keys: |
+                      ${{ runner.os }}-${{ matrix.build }}-rust-${{ hashFiles('**/rust-toolchain', '**/Cargo.toml*', '**/rustfmt.toml') }}-${{ steps.rust-setup.outputs.week }}
```

Comment:
This first key is not needed since it has been set in `key`

Reply:
Fitst tested it, but can't find the cache key.
https://github.com/black7375/emacs-ng/runs/2320293270
```  
Cache not found for input keys: Linux-webrender-rust-a04d7bed436a11b5e00681f64766cdcfa2548c7b424f71095a6d22a6e3a66677-2021y_04m_15w, Linux-webrender-rust-a04d7bed436a11b5e00681f64766cdcfa2548c7b424f71095a6d22a6e3a66677-, Linux-webrender-rust
```

If you delete the key, an error will occur.
https://github.com/black7375/emacs-ng/runs/2320309690?check_suite_focus=true
```
Error: Input required and not supplied: key
```

Current policy seems to be correct.